service/matching/tasklist/task_list_manager.go

```diff
@@ -491,10 +497,23 @@ func (c *taskListManagerImpl) AddTask(ctx context.Context, params AddTaskParams)
 	return syncMatch, err
 }
 
-// DispatchTask dispatches a task to a poller. When there are no pollers to pick
-// up the task or if rate limit is exceeded, this method will return error. Task
-// *will not* be persisted to db
+// DispatchTask dispatches a task to a poller on the active side. When there are no pollers to pick
+// up the task or if the rate limit is exceeded, this method will return error. Task
+// *will not* be persisted to db. On the passive side, dispatches the task to the taskCompleter; it will attempt
+// to complete the task if it has already been started.
 func (c *taskListManagerImpl) DispatchTask(ctx context.Context, task *InternalTask) error {
+	// optional configuration to enable cleanup of tasks, in the standby cluster, that have already been started
+	if c.config.EnableStandbyTaskCompletion() {
+		if err := c.taskCompleter.CompleteTaskIfStarted(ctx, task); err != nil {
```

Comment:
Stylistic nit: this is maybe a little surprising a pathway to look at if you're not sure about the active/passive split. What do you think about making it a little more explicit? 

eg:
```
if c.isDomainActive(ctx, task) {
    return c.matcher.MustOffer(ctx, task)
}

// everything hereafter you can be reasonably sure is passive path
if c.config.EnableStandbyTaskComlpletion() {
    ...
}
```

Reply:
do you mean something like this

```
func (c *taskListManagerImpl) DispatchTask(ctx context.Context, task *InternalTask) error {
	// check if this is the active cluster for the domain
	domainEntry, err := c.domainCache.GetDomainByID(task.Event.TaskInfo.DomainID)
	if err != nil {
		return fmt.Errorf("unable to fetch domain from cache: %w", err)
	}

	if _, err = domainEntry.IsActiveIn(c.clusterMetadata.GetCurrentClusterName()); err == nil {
		return c.matcher.MustOffer(ctx, task)
	}
	
	// optional configuration to enable cleanup of tasks, in the standby cluster, that have already been started
	if c.config.EnableStandbyTaskCompletion() {
		if err := c.taskCompleter.CompleteTaskIfStarted(ctx, task); err != nil {
			if errors.Is(err, errDomainIsActive) {
				return c.matcher.MustOffer(ctx, task)
			}
			return err
		}

		return nil
	}

	return c.matcher.MustOffer(ctx, task)
}
```

Reply:
yeah exactly, it just means it's very clear what's running on standby only